### ISSUE TITLE
Crashed Thread: AdobeMarketingMobile::StringUtils::Quotify on iOS

### ISSUE THREAD

**Issue by ynnckcmprnl** [Adobe-Marketing-Cloud/acp-sdks#477]
Hi,

After updating the SDK to the latest version since having reported issue #473, we're seeing a different crash happening a lot. We've come to the point that crashes in the Adobe SDK are our most common crashes in Crashlytics. 

All the crashes happened with the app in the background and on all iOS versions. 
Any help or tips?

```
PODS:
  - ACPAnalytics (2.4.0):
    - ACPAnalytics/main (= 2.4.0)
    - ACPCore (>= 2.5.0)
  - ACPAnalytics/main (2.4.0):
    - ACPCore (>= 2.5.0)
  - ACPCore (2.8.1):
    - ACPCore/main (= 2.8.1)
  - ACPCore/main (2.8.1)
  - ACPTarget (2.1.6):
    - ACPCore (>= 2.5.0)
    - ACPTarget/iOS (= 2.1.6)
  - ACPTarget/iOS (2.1.6):
    - ACPCore (>= 2.5.0)
```

```
Crashed: Thread #1
EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x2ca100000021a218

Crashed: Thread
0 0x105ab0220 AdobeMarketingMobile::StringUtils::Quotify(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned long) + 4355473952
1  0x105a4c3dc AdobeMarketingMobile::Event::ToStringImpl(AdobeMarketingMobile::ToStringOptions const&) const + 4355064796
2 0x105a6dbe4 AdobeMarketingMobile::ToStringImpl(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, AdobeMarketingMobile::ObjectInterface const&, AdobeMarketingMobile::ToStringOptions const&) + 4355202020
3 0x105a6dc8c void AdobeMarketingMobile::StringUtils::ToString<AdobeMarketingMobile::ObjectInterface>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, AdobeMarketingMobile::ObjectInterface const&, AdobeMarketingMobile::ToStringOptions const&) + 4355202188
4  0x105a6da1c std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > AdobeMarketingMobile::StringUtils::ToString<AdobeMarketingMobile::ObjectInterface>(AdobeMarketingMobile::ObjectInterface const&, AdobeMarketingMobile::ToStringOptions const&) + 4355201564
5  x105a5559c AdobeMarketingMobile::EventHubInner::ProcessEvent(std::__1::list<std::__1::shared_ptr<AdobeMarketingMobile::Module>, std::__1::allocator<std::__1::shared_ptr<AdobeMarketingMobile::Module> > > const&, std::__1::shared_ptr<AdobeMarketingMobile::Event>&) + 4355102108
6 0x105a55380 AdobeMarketingMobile::EventHubInner::DoEvent(std::__1::shared_ptr<AdobeMarketingMobile::Event>, std::__1::shared_ptr<AdobeMarketingMobile::Module> const&, std::__1::shared_ptr<AdobeMarketingMobile::EventData> const&, bool) + 4355101568
7 0x105a58f5c std::__1::__function::__func<AdobeMarketingMobile::EventHubInner::Dispatch(std::__1::shared_ptr<AdobeMarketingMobile::Event> const&)::$_2, std::__1::allocator<AdobeMarketingMobile::EventHubInner::Dispatch(std::__1::shared_ptr<AdobeMarketingMobile::Event> const&)::$_2>, void ()>::operator()() + 4355116892
8 0x105a6fbf8 AdobeMarketingMobile::TryCatch(std::__1::function<void ()> const&, std::__1::function<void (AdobeMarketingMobile::SdkError&)> const&) + 4355210232
9 0x105a6c668 AdobeMarketingMobile::(anonymous namespace)::ProxyTaskExecutor::ParentTask(std::__1::shared_ptr<AdobeMarketingMobile::(anonymous namespace)::ProxyTaskExecutorData> const&) + 4355196520
10 0x1059e3044 std::__1::__function::__func<AdobeMarketingMobile::(anonymous namespace)::NativeTaskExecutor::AddTaskImpl(std::__1::function<void ()>, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, AdobeMarketingMobile::SdkError&)>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::$_0, std::__1::allocator<AdobeMarketingMobile::(anonymous namespace)::NativeTaskExecutor::AddTaskImpl(std::__1::function<void ()>, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, AdobeMarketingMobile::SdkError&)>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::$_0>, void ()>::operator()() + 4354633796
11 0x105a6fbf8 AdobeMarketingMobile::TryCatch(std::__1::function<void ()> const&, std::__1::function<void (AdobeMarketingMobile::SdkError&)> const&) + 4355210232
12 0x105a73c04 std::__1::__function::__func<AdobeMarketingMobile::WorkerMain(std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorData> const&, std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorWorkerData> const&)::$_2::operator()() const::'lambda0'(), std::__1::allocator<AdobeMarketingMobile::WorkerMain(std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorData> const&, std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorWorkerData> const&)::$_2::operator()() const::'lambda0'()>, void ()>::operator()() + 4355226628
13 0x105a6fbf8 AdobeMarketingMobile::TryCatch(std::__1::function<void ()> const&, std::__1::function<void (AdobeMarketingMobile::SdkError&)> const&) + 4355210232
14 0x105a6fb48 AdobeMarketingMobile::TrySwallow(std::__1::function<void ()> const&) + 4355210056
15 0x105a731d0 std::__1::__function::__func<AdobeMarketingMobile::WorkerMain(std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorData> const&, std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorWorkerData> const&)::$_2, std::__1::allocator<AdobeMarketingMobile::WorkerMain(std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorData> const&, std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorWorkerData> const&)::$_2>, void ()>::operator()() + 4355224016
16 0x105a6fbf8 AdobeMarketingMobile::TryCatch(std::__1::function<void ()> const&, std::__1::function<void (AdobeMarketingMobile::SdkError&)> const&) + 4355210232
17 0x105a6fb48 AdobeMarketingMobile::TrySwallow(std::__1::function<void ()> const&) + 4355210056
18 0x105a72c94 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, AdobeMarketingMobile::CreateWorkerThread(std::__1::shared_ptr<AdobeMarketingMobile::TaskExecutorData> const&)::$_1> >(void*) + 4355222676
19 libsystem_pthread.dylib        0x1e8bb4b3c _pthread_start + 288
20 libsystem_pthread.dylib        0x1e8bb9880 thread_start + 8
```

**Comment by shalehaha** [comment]
@ynnckcmprnl 
Thanks for reporting this, we will take a look and let you know once we have a fix for it.

**Comment by shalehaha** [comment]
@ynnckcmprnl 
what's the log level you are using for the production app? `Verbose`?

**Comment by ynnckcmprnl** [comment]
Indeed, the loglevel was set to verbose.

`ACPCore.setLogLevel(.verbose)`

**Comment by shalehaha** [comment]
@ynnckcmprnl 
For performance sake, it is always better to use `warning` or `error` for the production app, and it will also help you avoid this crash before we have a fix for it.

**Comment by ynnckcmprnl** [comment]
I was also surprised to see it was still set to verbose.

We'll change the log level in the next update.

**Comment by shalehaha** [comment]
Fixed in 2.9.1 release
